src/libraries/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -138,7 +138,7 @@ void ICollection.CopyTo(Array array, int arrayIndex)
                 throw new ArgumentException(SR.Arg_NonZeroLowerBound, nameof(array));
             }
 
-            if (arrayIndex < 0 || arrayIndex > array.Length)
```

Comment:
There are lots of such patterns in the code, makes sense to revise them all (some of them can be found in this list: https://gist.github.com/EgorBo/e6bbd36a2d87975fd3c9763f7a630bc9 - it also includes this Stack.CopyTo case).